logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -394,15 +396,21 @@ class ConversationDataSource(
             when (conversation.protocol) {
                 is Conversation.ProtocolInfo.Proteus ->
                     wrapApiRequest {
-                        conversationApi.removeConversationMember(idMapper.toApiModel(userID), idMapper.toApiModel(conversationId))
+                        conversationApi.removeMember(idMapper.toApiModel(userID), idMapper.toApiModel(conversationId))
                     }.fold({
                         Either.Left(it)
                     }, {
-                        wrapStorageRequest {
-                            conversationDAO.deleteMemberByQualifiedID(
-                                idMapper.toDaoModel(userID),
-                                idMapper.toDaoModel(conversationId)
-                            )
+                        when (it) {
+                            // TODO I think on unchanged we should also remove member from local storage
+                            // probably there was some synchronisation issue that this user was already removed
+                            ConversationMemberChangeDTO.Unchanged -> Either.Right(Unit)
+                            is ConversationMemberChangeDTO.Added -> wrapStorageRequest {
```

Comment:
how come a delete request could return with a member added response?

Reply:
Refactored to two undependable models